dcos/jsonitem.py

```diff
@@ -283,8 +283,9 @@ def _parse_url(value):
     if value_regex is None:
         scheme_match = re.match(scheme_pattern, value, re.IGNORECASE)
         if scheme_match is None:
-            msg = 'Please check url {!r}. Missing http(s)://'.format(value)
-            raise DCOSException(msg)
+            logger.debug("Defaulting URL to https scheme")
+            value = "https://" + value.lstrip(scheme_pattern)
```

Comment:
IMO the correct way to tackle this would be to use `urllib.parse.urlparse()`, to use `netloc` and `scheme` from the parse result tuple, and to re-assemble the URL using `urllib.parse.unparse()`.

That would get rid of all the regex craziness here.